cmd/clusterctl/client/cluster/components.go

```diff
@@ -209,6 +215,12 @@ func (p *providerComponents) Delete(options DeleteOptions) error {
 			continue
 		}
 
+		if obj.GetKind() == deploymentKind {
+			if err := scaleDownDeployment(ctx, cs, obj); err != nil {
```

Comment:
Let's document why we are doing this

Reply:
Absolutely, this wasn't meant to be ready for review yet :)